stixcore/processing/publish.py

```diff
@@ -0,0 +1,300 @@
+
+    args = parser.parse_args(args)
+
+    db_file = Path(args.db_file)
+    fits_dir = Path(args.fits_dir)
+    if not fits_dir.exists():
+        logger.error(f'path not found to input files: {fits_dir}')
+        return
+
+    wait_period = u.Quantity(args.waiting_period)
+    wait_period_s = wait_period.to(u.s).value
+
+    target_dir = Path(args.target_dir)
+    if not target_dir.exists():
+        logger.info(f'path not found to target dir: {target_dir} creating dir')
+        target_dir.mkdir(parents=True, exist_ok=True)
+
+    include_levels = dict([(level, 1) for level in
+                           args.include_levels.lower().replace(" ", "").split(",")])
+
+    if args.include_versions == "*":
+        include_all_versions = True
+    else:
+        include_all_versions = False
+        include_versions = dict([(int(version), 1) for version in
+                                 args.include_versions.lower().replace(" ", "")
+                                 .replace("v", "").split(",")])
+
+    if args.include_products == "*":
+        include_all_products = True
+    else:
+        include_all_products = False
+        include_products = dict([(prod, 1) for prod in
+                                 args.include_products.lower().replace(" ", "").split(",")])
+
+    candidateds = fits_dir.rglob("solo_*.fits")
+    to_publish = list()
+    now = datetime.now().timestamp()
+    hist = PublishHistoryStorage(db_file)
+
+    for c in candidateds:
+        parts = c.name[:-5].split('_')
+        level = parts[1].lower()
+
+        if not include_all_versions:
+            version = int(parts[4].lower().replace("v", ""))
+            if version not in include_versions:
+                continue
+
+        last_mod = c.stat().st_mtime
+
+        # should the level by published
+        if level not in include_levels:
+            continue
+
+        # is the waiting time after last modification done
+        if (now - last_mod) < wait_period_s:
+            continue
+
+        if not include_all_products:
+            product = str(parts[2].lower().split("-")[1])
+            if product not in include_products:
+                continue
+
+        old = hist.find_by_name(c.name)
+
+        # was the same file already published
+        if len(old) == 1:
+            old = old[0]
+            if old['m_date'] == last_mod:
+                continue
+
+        to_publish.append(c)
+
+    published = defaultdict(list)
+    for p in to_publish:
+        try:
+            add_res, data = hist.add(p)
+            shutil.copy(p, target_dir)
+            published[add_res].append(data)
+            logger.info(f"published file: {data} > {str(add_res)}")
+        except Exception as e:
+            published['failed'].append((p, e))
+            logger.error(e, exc_info=True)
+
+    send_mail_report(published)
+    hist.close()
+
+
+if __name__ == '__main__':
+    publish_fits_to_esa(sys.argv[1:])
```

Comment:
So would the plan to be to run this a cron job or similar?

Reply:
yes